### PR TITLE
[FLINK-36711][runtime] Add tips when System out and err are redirected to LOG or discarded

### PR DESCRIPTION
**Pull request by Vincent-Woo**
## What is the purpose of the change

When System.out and System.err are redirected to LOG or discarded, there are some scenarios that do not match the user's expectations, leading them to be unclear about how to view the relevant output information. Add tips when System out and err are redirected to LOG or discarded to remind the person who going to view the out/err file of the expected effect of the current configuration.(See [FLINK-36711](https://issues.apache.org/jira/browse/FLINK-36711) for details)

## Brief change log

  - Add tips when System out and err are redirected to LOG or discarded

## Verifying this change

This change added tests and can be verified as follows:
  - Updated SystemOutRedirectionUtilsTest to accommodate this change

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by Vincent-Woo** [comment]
@flinkbot run azure

**Comment by Vincent-Woo** [comment]
@flinkbot run azure

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/SystemOutRedirectionUtilsTest.java [hunk 1]
```diff
@@ -77,26 +76,42 @@ void testDefaultSystemOutAndErr() {
...
+        String logContext = "This is log context!";
+        System.out.print(logContext);
+        assertThat(outStream.toString()).isEqualTo(SystemOutRedirectionUtils.OUT_TO_LOG_TIPS);
+
+        System.err.print(logContext);
+        assertThat(errStream.toString()).isEqualTo(SystemOutRedirectionUtils.ERR_TO_LOG_TIPS);
+    }
+
+    @Test
+    void testSystemOutAndErrAreRedirectedToIgnore() {
```
Comment (1996fanrui) [inline flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/SystemOutRedirectionUtilsTest.java:99]: We don't need 2 test methods here. The `SystemOutMode`, out tips and err tips could be argument.

You could refer to `DynamicFilteringITCase#testReuseDimSide` test[1], it uses 2 arguments. And `parameters` method [2]as the argument provider.

[1] https://github.com/apache/flink/blob/77e7ac076c35d229fab89b8b6ae15138b0e41a22/flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/batch/sql/DynamicFilteringITCase.java#L245
[2] https://github.com/apache/flink/blob/77e7ac076c35d229fab89b8b6ae15138b0e41a22/flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/batch/sql/DynamicFilteringITCase.java#L49

**Review by 1996fanrui — COMMENTED** [review]
Thanks @Vincent-Woo for the improvment!

Overall LGTM, I left 2 minor comments, please take a look in your free time, thanks~

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/SystemOutRedirectionUtilsTest.java (hunk 1 above)
Comment (Vincent-Woo) [reply flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/SystemOutRedirectionUtilsTest.java]: @1996fanrui Thanks for the guidance, I made some changes, PTAL

File: flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/SystemOutRedirectionUtils.java [hunk 2]
```diff
@@ -40,6 +40,24 @@ public class SystemOutRedirectionUtils {
...
+            "Tips: System.err is redirected to LOG.error as configured."
+                    + " View the log file and search [SystemOutRedirectionUtils] for output.\n";
+
+    @VisibleForTesting
+    static final String OUT_IGNORE_TIPS =
+            "Tips: System.out will be directly ignored as configured.\n";
+
+    @VisibleForTesting
+    static final String ERR_IGNORE_TIPS =
+            "Tips: System.err will be directly ignored as configured.\n";
```
[the remark this answers is not in the document]
Comment (Vincent-Woo) [reply flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/SystemOutRedirectionUtils.java]: Good idea! I dealt with it.

**Review by 1996fanrui — APPROVED** [review]
Thanks @Vincent-Woo for the quick update!

LGTM assuming the CI is green.

**Comment by Vincent-Woo** [comment]
@flinkbot run azure

**Comment by Vincent-Woo** [comment]
@flinkbot run azure

**Pull request merged**